text/0000-local-snapshot-file-format/0000-local-snapshot-file-format.md

```diff
@@ -0,0 +1,147 @@
++ Feature name: `local_snapshot_file_format`
++ Start date: 2020-08-25
++ RFC PR: [iotaledger/protocol-rfcs#0000](https://github.com/iotaledger/protocol-rfcs/pull/0000)
+
+# Summary
+
+This RFC defines a file format for local snapshots which is compatible with Chrysalis Phase 2.
+
+# Motivation
+
+Nodes create local snapshots to produce ledger representations at a point in time of a given milestone in order to be able to:
+* start up from a recent milestone instead of having to synchronize from the genesis transaction.
+* delete old transaction data below a given milestone.
+
+Current node implementations use a [local snapshot file format](https://github.com/iotaledger/iri-ls-sa-merger/tree/351020d3b5e342b6e9a41f2868575ab7ff8c251c#generating-an-export-file-from-a-localsnapshots-db) which only works with account based ledgers. For Chrysalis Phase 2 this file format has to be assimilated to support a UTXO based ledger.
+
+# Detailed design
+
+Since a UTXO based ledger is much larger in size, this RFC proposes two formats for snapshot files: 
+* a `full` format which represents a complete ledger state.
+* a `delta` format which only contains diffs (consumed and spent outputs) of milestones from a given milestone index onwards.
+
+This separation allows nodes to swiftly create new delta snapshot files, which then can be distributed with a companion full snapshot file to reconstruct a recent state.
+
+Unlike the current format, this new formats do not longer include spent addresses since this information is no longer held by nodes.
```

Comment:
```suggestion
# Detailed design

Since a UTXO based ledger is much larger in size, this RFC proposes two formats for snapshot files: 
* A `full` format which represents a complete ledger state.
* A `delta` format which only contains diffs (consumed and spent outputs) of milestones from a given milestone index onwards.

This separation allows nodes to swiftly create new delta snapshot files, which then can be distributed with a companion full snapshot file to reconstruct a recent state.

Unlike the current format, these new formats do not include spent addresses since this information is no longer held by nodes.
```